src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/RemoteProjectSnapshot.cs

```diff
@@ -198,6 +209,17 @@ private RazorConfiguration CreateRazorConfiguration()
             razorLanguageVersion = RazorLanguageVersion.Latest;
         }
 
-        return new(razorLanguageVersion, configurationName, Extensions: [], UseConsolidatedMvcViews: true);
+        var compilation = await _project.GetCompilationAsync().ConfigureAwait(false);
+
+        var suppressAddComponentParameter = compilation is null
+            ? false
+            : !compilation.HasAddComponentParameter();
+
+        return new(
+            razorLanguageVersion,
+            configurationName,
+            Extensions: [],
+            SuppressAddComponentParameter: suppressAddComponentParameter,
+            UseConsolidatedMvcViews: true);
```

Comment:
nit: call with arguments in the correct order to get rid of a redundant named argument.

```suggestion
            UseConsolidatedMvcViews: true,
            suppressAddComponentParameter);
```